packages/editor/src/bindings/post-meta.js

```diff
@@ -7,22 +7,43 @@ import { store as coreDataStore } from '@wordpress/core-data';
  * Internal dependencies
  */
 import { store as editorStore } from '../store';
+import { unlock } from '../lock-unlock';
+
+function getMetadata( registry, context ) {
+	let metaFields = {};
+	const { type } = registry.select( editorStore ).getCurrentPost();
+	const { getEditedEntityRecord } = registry.select( coreDataStore );
+	const { getRegisteredPostMeta } = unlock(
+		registry.select( coreDataStore )
+	);
+
+	if ( type === 'wp_template' ) {
+		const fields = getRegisteredPostMeta( context?.postType );
+		// Populate the `metaFields` object with the default values.
+		Object.entries( fields || {} ).forEach( ( [ key, props ] ) => {
+			metaFields[ key ] = props.default;
+		} );
+	} else {
+		metaFields = getEditedEntityRecord(
```

Comment:
When compared `getRegisteredPostMeta` with `getEditedEntityRecord` calls on the same post type, how much the meta fields returned can differ?

Reply:
`getRegisteredPostMeta` returns the fields definition. This means it includes the `label`, `description`, and `default` BUT not the value specific for that post ID. We can use it for templates.

In the cases where we have a post ID, we need to call `getEditedEntityRecord`, which includes the custom field value for that specific post.